### DISCUSSION TITLE
Stuck in safeboot after Safeboot partition migration on Shelly Plus 1

### DISCUSSION THREAD

**Discussion by xpfgsyb** [arendst/Tasmota#19414] [Q&A] [answered]
Hi all, I got a new Shelly Plus 1 and installed Tasmota 12.5.0 on it via the OTA process as described on https://github.com/tasmota/mgos32-to-tasmota32.

After that I tried to convert it to the safeboot partition scheme via the Partition Wizard. The device rebooted several times during the process and I left it alone for about 10 minutes. After that I'm welcomed by Tasmota 13.1.0 Safeboot and I see no way to switch to the production version. Restarting it via the GUI or power it off and on again always boot in the safeboot mode again. Restart 3 also boot into the safeboot mode. Switching modules via module 0 or module 1 also boot into the safeboot mode.

On the device information page I can see that it changed the partition scheme:
Partition safeboot*: 832 KB (used 95%)
Partition app0: 2688 KB (used 29%)
Partition fs: 448 KB

I have no idea how to get to the production partition from here and I hope that someone here has a great idea :)

**Comment by sfromis** [comment]
IDK what happened there, but the app0 partition content looks way off in size, 29% of 2688 KB is 780 KB, a lot smaller than what should be there. Maybe it would help issuing these commands:
```
OtaUrl http://ota.tasmota.com/tasmota32/tasmota32.bin
Upgrade 1
```
As the partition sizes for safeboot and app0 looks right, it may just be the final upgrade missing, but I'm not sure of anything, or even if my suggestion is useful....

**Comment by xpfgsyb** [comment]
Thanks, I tried that and the percentage looks better now (app0 with 68% used), but I'm still stuck in safeboot:
Restart -> safeboot
Restart 3 -> safeboot
Module 0 was active, I therefore tried "Module 1" -> safeboot and still in module 0 afterwards:
```
16:37:46.654 CMD: modules
16:37:46.657 RSL: RESULT = {"Modules":{"0":"Shelly Plus 1","1":"ESP32-DevKit"}}
16:37:50.215 CMD: module
16:37:50.218 RSL: RESULT = {"Module":{"0":"Shelly Plus 1"}}
```
Power off and on again -> safeboot

Looks a bit like it just don't want to boot into module 1 which would be the prod environment I guess.

**Reply by sfromis** [reply]
While booted in safeboot, you cannot modify any settings, including `Module`. However, this setting is not about "prod environment", but instead to configure gpio pins, and this should not be related to the success (or not) of switching to the regular boot partition.

Anyway, I forgot about one detail, that the Shelly Plus 1 is probably with a single core ESP32 variant. Try upgrading to:
`tasmota32solo1.bin`

**Reply by xpfgsyb** [reply]
Yes, the information page says it's a single-core, so I tried the tasmota32solo1.bin too.
The result is the same unfortunately, it's still booting into safeboot. Restart, Restart 3 or power off/on doesn't make a difference.

**Comment by xpfgsyb** [comment, marked as answer]
To update this question: I found no other solution and had to reflash it via the serial interface.

**Reply by peepshow-21** [reply]
same here. full re-install needed.

**Comment by NickUfer** [comment]
This is just great. Stuck in safeboot without internet connectivity and nobody had the idea that this might happen and there is no F'ing way to reflash because it was built by some company... great stuff. Why is everyone using this when you constantly can destroy your device.
How the F*** is there no way to change the wifi config in safeboot mode. This is ridiculous.

**Reply by sfromis** [reply]
As long as you can contact the device, you can revert to normal boot using the command `Restart 3`.

Or you can do an upgrade by local file upload, as long as you get the binary file.

**Reply by jaburges** [reply]
a little strong - BUT - changing the WifFi should be a requirement in safeboot.

**Comment by ok11** [comment]
Having the same problem. Not sure why the issue was closed. Sure, serial reflashing is always possible, but it seems there is a bug here...
